### PR TITLE
Update 3.1 to follow BASHER and Zero Trust guidelines

### PR DESCRIPTION
**Pull request by derisen**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 3-Authorization-II/1-call-api/AppCreationScripts/sample.json [hunk 1]
```diff
@@ -1,80 +1,113 @@
 {
   "Sample": {
-    "Title": "Angular single-page application that authenticates users with Azure AD and calls a protected .NET Core web API",
-    "Level": 200,
-    "Client": "Angular SPA",
-    "Service": ".NET Core web API",
-    "RepositoryUrl": "ms-identity-javascript-angular-tutorial",
-    "Endpoint": "AAD v2.0"
+      "Author": "derisen",
+      "Title": "Angular single-page application that authenticates users with Azure AD and calls a protected .NET Core web API",
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/AppCreationScripts/sample.json:4]: >Angular single-page application that authenticates users with Azure AD and calls a protected .NET Core web API [](http://example.com/codeflow?start=16&length=110)

An Angular SPA using MSAL Angular to sign-in users with Azure Active Directory and call a protected .NET Core web API

File: 3-Authorization-II/1-call-api/AppCreationScripts/README_STEPS.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
+---
+page_type: sample
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/AppCreationScripts/README_STEPS.md:2]: >sample [](http://example.com/codeflow?start=11&length=6)

remove this from commit

File: 3-Authorization-II/1-call-api/README.md [hunk 3]
```diff
@@ -15,12 +15,13 @@
 
 ## Overview
 
-This sample demonstrates an Angular single-page application (SPA) calling a ASP.NET Core web API secured with [Azure Active Directory](https://docs.microsoft.com/azure/active-directory/fundamentals/active-directory-whatis) (Azure AD) using the [Microsoft Authentication Library for Angular](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-angular) (MSAL Angular) for the SPA and the [Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) (M.I.W) for the web API.
+This sample demonstrates an Angular single-page application (SPA) calling a ASP.NET Core web API secured with [Azure Active Directory](https://docs.microsoft.com/azure/active-directory/fundamentals/active-directory-whatis) (Azure AD) using the [Microsoft Authentication Library for Angular](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-angular) (MSAL Angular) for the SPA and the [Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) (MIW) for the web API.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:18]: >https://docs.microsoft.com/azure/active-directory/fundamentals/active-directory-whatis [](http://example.com/codeflow?start=135&length=86)

aka.ms/identityplatform

Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:18]: >calling [](http://example.com/codeflow?start=66&length=7)

signing-in a user and calling

File: 3-Authorization-II/1-call-api/README.md [hunk 4]
```diff
@@ -15,12 +15,13 @@
...
 
 ## Scenario
 
 1. The client Angular SPA uses **MSAL Angular** to sign-in and obtain a JWT access token from **Azure AD**.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:22]: >access token [](http://example.com/codeflow?start=76&length=12)

aka.ms/access-tokens

**Comment by kalyankrishna1** [comment]
> - An **Azure AD** tenant. For more information see: [How to get an Azure AD tenant](https://docs.microsoft.com/azure/active-directory/develop/quickstart-create-new-tenant)

this is an outdated link

---
Refers to: 3-Authorization-II/1-call-api/README.md:40 in 4a60819. [](commit_id = 4a608195002fd51d5c5b6f842b617246ef1fff5b, deletion_comment = False)

File: 3-Authorization-II/1-call-api/README.md [hunk 5]
```diff
@@ -30,8 +31,9 @@ This sample demonstrates an Angular single-page application (SPA) calling a ASP.
 |-------------------------------------|------------------------------------------------------------|
 | `SPA/src/app/auth-config.ts`        | Authentication parameters for SPA project reside here.     |
 | `SPA/src/app/app.module.ts`         | MSAL Angular is initialized here.                          |
-| `API/appsettings.json`              | Authentication parameters for API project reside here.     |
-| `API/Startup.cs`                    | Microsoft.Identity.Web is initialized here.                |
+| `API/TodoListAPI/appsettings.json`  | Authentication parameters for API project reside here.     |
+| `API/TodoListAPI/Startup.cs`        | Microsoft.Identity.Web is initialized here.                |
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:35]: >Microsoft.Identity.Web is initialized here. [](http://example.com/codeflow?start=40&length=43)

Microsoft.Identity.Web is initialized to protect the web API here.

File: 3-Authorization-II/1-call-api/README.md [hunk 6]
```diff
@@ -30,8 +31,9 @@ This sample demonstrates an Angular single-page application (SPA) calling a ASP.
 |-------------------------------------|------------------------------------------------------------|
 | `SPA/src/app/auth-config.ts`        | Authentication parameters for SPA project reside here.     |
 | `SPA/src/app/app.module.ts`         | MSAL Angular is initialized here.                          |
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:33]: >MSAL Angular is initialized here [](http://example.com/codeflow?start=40&length=32)

MSAL Angular to sign-in user is initialized here

File: 3-Authorization-II/1-call-api/README.md [hunk 7]
```diff
@@ -121,37 +124,47 @@ There are two projects in this sample. Each needs to be separately registered in
...
+The first thing that we need to do is to declare the unique [resource](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow) URI that the clients will be using to obtain access tokens for this Api. To declare an resource URI, follow the following steps:
+   - Select `Set` next to the **Application ID URI** to generate a URI that is unique for this app.
+   - For this sample, accept the proposed Application ID URI (`api://{clientId}`) by selecting **Save**.
+1. All APIs have to publish a minimum of two [scopes](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code), also called [delegated permissions](https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#permission-types), for the client's to obtain an access token successfully. To publish a scope, follow these steps:
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:131]: >f two [scopes [](http://example.com/codeflow?start=39&length=13)

the statement is wrong and fixed in code gen.. re-generate

File: 3-Authorization-II/1-call-api/README.md [hunk 8]
```diff
@@ -121,37 +124,47 @@ There are two projects in this sample. Each needs to be separately registered in
...
+        - For **Scope name**, enter `TodoList.Read`.
+        - Select **Admins and users** options for **Who can consent?**.
+        - For **Admin consent display name** type `Access msal-dotnet-api`.
+        - For **Admin consent description** type `Allows the app to access msal-dotnet-api to read todo list.`
+        - For **User consent display name** type `Access msal-dotnet-api`.
+        - For **User consent description** type `Allows the app to access msal-dotnet-api to read todo list.`
+        - Keep **State** as **Enabled**.
+        - Select the **Add scope** button on the bottom to save this scope.
+   - Repeat the steps above for publishing another scope named `TodoList.ReadWrite`.
+1. APIs should also publish scopes that can only be consumed by applications (not users), also known as [application permissions](https://docs.microsoft.com/azure/active-directory/develop/permissions-consent-overview#types-of-permissions). To do so, select the **App roles** blade to the left.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:142]: >scopes [](http://example.com/codeflow?start=28&length=6)

permissions (not scopes), please check codegen for me. @aremo-ms FYI

File: 3-Authorization-II/1-call-api/README.md [hunk 9]
```diff
@@ -198,80 +211,141 @@ In a separate console window, execute the following commands:
...
-            // Creating policies that wraps the authorization requirements
-            services.AddAuthorization();
-        }
+public void ConfigureServices(IServiceCollection services)
+{
+    // This is required to be instantiated before the OpenIdConnectOptions starts getting configured.
+    // By default, the claims mapping will map claim names in the old format to accommodate older SAML applications.
+    // 'http://schemas.microsoft.com/ws/2008/06/identity/claims/role' instead of 'roles'
+    // This flag ensures that the ClaimsIdentity claims collection will be built from the claims in the token
+    JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:254]: you can drop this, the line for "AddMicrosoftIdentityWebApiAuthentication" is all we need here

**Comment by kalyankrishna1** [comment]
> - A user account in your **Azure AD** tenant. This sample will not work with a **personal Microsoft account**. Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a personal account and have never created a user account in your directory before, you need to do that now.

add tow lines,
1. get latest NPM
1. get latest .NET core 

---
Refers to: 3-Authorization-II/1-call-api/README.md:41 in 4a60819. [](commit_id = 4a608195002fd51d5c5b6f842b617246ef1fff5b, deletion_comment = False)

File: 3-Authorization-II/1-call-api/SPA/src/app/auth-config.ts [hunk 10]
```diff
@@ -10,21 +10,26 @@ import { LogLevel, Configuration, BrowserCacheLocation } from '@azure/msal-brows
...
+ * Configuration object to be passed to MSAL instance on creation.
  * For a full list of MSAL.js configuration parameters, visit:
- * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/configuration.md 
+ * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/configuration.md
  */
 export const msalConfig: Configuration = {
   auth: {
     clientId: 'Enter_the_Application_Id_Here', // This is the ONLY mandatory field that you need to supply.
     authority: 'https://login.microsoftonline.com/Enter_the_Tenant_Info_Here', // Defaults to "https://login.microsoftonline.com/common"
     redirectUri: '/', // Points to window.location.origin. You must register this URI on Azure portal/App Registration.
```
Comment (salman90) [inline 3-Authorization-II/1-call-api/SPA/src/app/auth-config.ts:21]: You might want to add a redirect page to follow the MSAL best practices

File: 3-Authorization-II/1-call-api/README.md [hunk 11]
```diff
@@ -198,80 +211,141 @@ In a separate console window, execute the following commands:
...
+    services.AddMicrosoftIdentityWebApiAuthentication(Configuration);
+
+    // ...
+}
 ```
 
-In your controller, add [Authorize] decorator, which will make sure all incoming requests have an authentication bearer:
+For validation and debugging purposes, developers can decode **JWT**s (*JSON Web Tokens*) using [jwt.ms](https://jwt.ms).
+
+### Verifying permissions
```
Comment (kalyankrishna1) [inline 3-Authorization-II/1-call-api/README.md:265]: brilliant, put this section in Codegen too !

File: 3-Authorization-II/1-call-api/README.md [hunk 12]
```diff
@@ -198,80 +211,141 @@ In a separate console window, execute the following commands:
 ## Explore the sample
 
 1. Open your browser and navigate to `http://localhost:4200`.
-2. Sign-in using the button on the top-right corner.
-3. Select the **TodoList** button on the navigation bar to access your todo list.
+1. Select the **Sign In** button on the top right corner. Choose either **Popup** or **Redirect** flows.
```
Comment (salman90) [inline 3-Authorization-II/1-call-api/README.md:214]: Add some pictures, please

**Review by salman90 — COMMENTED** [review]
Looks good!

File: 3-Authorization-II/1-call-api/SPA/src/app/auth-config.ts (hunk 10 above)
Comment (derisen) [reply 3-Authorization-II/1-call-api/SPA/src/app/auth-config.ts]: done

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

**Pull request merged**